sidestep_engine/data/lyrics_provider_genius.py

```
@@ -38,6 +39,38 @@ def _is_wrapped_encoding_error(exc: Exception) -> bool:
+    ``(Official Audio)``, ``(Remix)`` etc. that confuse Genius search.
+    """
+    # Remove parenthetical/bracketed content containing feature tags
+    # or common noise like "Official Audio", "Lyric Video", etc.
+    title = re.sub(
+        r"\s*[\(\[]\s*"
+        r"(?:w[/\s]|feat\.?\s|ft\.?\s|with\s|prod\.?\s|official|lyric|audio|video|remix|live|acoustic)"
+        r"[^)\]]*[\)\]]",
+        "", title, flags=re.IGNORECASE,
+    )
+    # Also strip trailing " - Topic" or " - Remix" style suffixes
+    title = re.sub(r"\s*[-–—]\s*(?:official|lyric|audio|video|topic).*$", "", title, flags=re.IGNORECASE)
```

Ruff (0.15.7)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)